src/main/java/com/letsparty/web/controller/PartyController.java

```diff
@@ -181,34 +191,89 @@ public String modify(@PathVariable int partyNo, @Valid PartyForm partyForm, Bind
 	}
 	
 	@PreAuthorize("isAuthenticated()")
-	@GetMapping("/{partyNo}/post")
+	@GetMapping("/{partyNo}/post-form")
 	public String addPost(@PathVariable int partyNo, @AuthenticationPrincipal LoginUser loginUser, Model model) {
 		UserPartyApplication upa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, loginUser.getId());
 		model.addAttribute("upa", upa);
 		PostForm postForm = new PostForm();
 		model.addAttribute("postForm", postForm);
-		return "/page/party/post";
+		return "/page/party/post-form";
 	}
 	
 	// 게시물 제출
 	@PreAuthorize("isAuthenticated()")
-	@PostMapping("/{partyNo}/post")
+	@PostMapping("/{partyNo}/post-form")
 	public String addPost(@PathVariable int partyNo, @AuthenticationPrincipal LoginUser loginUser, @Valid PostForm postForm,
 			BindingResult error, Model model) {
 		UserPartyApplication upa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, loginUser.getId());
 		model.addAttribute("upa", upa);
 		// TODO 제목, 본문 글자 수 제한하는 유효성 구현하기
 		if (error.hasErrors()) {
-			return "page/party/post";
+			return "page/party/post-form";
 		}
 		partyService.insertPost(postForm, partyNo, loginUser.getId());
 		return "redirect:/party/{partyNo}";
 	}
 	
 	@GetMapping("/{partyNo}")
-	public String home(@PathVariable int partyNo, Model model) {
-		// partyNo를 사용하여 파티 게시물을 조회합니다.
-		return "page/party/home";
+	public String home(@PathVariable int partyNo) {
+		// partyNo를 사용해서 마지막 게시글 번호를 불러온다.
+		int postNo = postService.getLastPostNoByPartyNo(partyNo);
+		return "redirect:/party/{partyNo}/read/" + postNo;
+	}
+	
+	@GetMapping("/{partyNo}/read/{postNo}")
+	public String read(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo) {
+		postService.readIncrement(partyNo, postNo);
+		return "redirect:/party/{partyNo}/post/{postNo}";
+	}
+	
+	@GetMapping("/{partyNo}/post/{postNo}")
+	public String readPost(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo, @AuthenticationPrincipal LoginUser loginUser, Model model) {
+		if (null != loginUser) {
+			User user = userService.getUserByName(loginUser.getId());
+			UserPartyApplication loginUpa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, loginUser.getId());
+			if (null != loginUpa) {
+				model.addAttribute("loginUpa", loginUpa);
+			}
+			model.addAttribute("loginUser", user);
+		}
```

Comment:
```suggestion
	public String readPost(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo, Principal principal, Model model) {
		if (principal != null && principal.getAuthorities().stream().noneMatch(authority -> authority.getAuthority().equals("ROLE_USER_PENDING"))) {
			UserPartyApplication loginUpa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, principal.getName());
			if (loginUpa != null) {
				model.addAttribute("loginUpa", loginUpa);
			}
			model.addAttribute("loginUserId", principal.getName());
		}
```

id만 필요하면 가볍게 Principal을 확인하는 게 좋은 것 같습니다.
대기자를 제외하는 코드를 추가하였습니다.

Reply:
일단 getAuthorities를 사용하려면 UserDetails를 사용하는 객체에서 써야 한다는데, 소셜로그인을 사용한 경우와 일반로그인을 한 경우에는 각각 다른 객체여서 위의 작성하신 코드를 어떻게 써야 할지 감이 안오네요.. 일단 이부분은 수정 없이 두겠습니다.

Reply:
Principal에 getAuthorities() 메서드가 없네요. 잘못 적었습니다.
```suggestion
	public String readPost(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo, Principal principal, Authentication authentication, Model model) {
		if (principal != null && authentication.getAuthorities().stream().noneMatch(authority -> authority.getAuthority().equals("ROLE_USER_PENDING"))) {
			UserPartyApplication loginUpa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, principal.getName());
			if (loginUpa != null) {
				model.addAttribute("loginUpa", loginUpa);
			}
			model.addAttribute("loginUserId", principal.getName());
		}
```
Authentication에 getAuthorities() 있으니 대체하면 될 것 같습니다.